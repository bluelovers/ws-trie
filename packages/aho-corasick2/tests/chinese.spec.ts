/**
 * Created by user on 2018/7/3/003.
 */

/// <reference types="mocha" />
/// <reference types="benchmark" />
/// <reference types="chai" />
/// <reference types="node" />

import { chai, relative, expect, path, assert, util, mochaAsync } from './_local-dev';

// @ts-ignore
import { ITest } from 'mocha';

import AhoCorasick from '..';

// @ts-ignore
describe(relative(__filename), () =>
{
	// @ts-ignore
	it(`Picks out chinese words`, function ()
	{
		let content, find_list, i, len, ref, word;

		let ac = new AhoCorasick();

		ref = ['五毛党', '国家', '黑手党', '党国', '毛五贴'];

		for (i = 0, len = ref.length; i < len; i++)
		{
			word = ref[i];
			ac.add(word);
		}
		ac.build_fail();

		find_list = ['毛五贴', '五毛党', '党国', '国家'];
		content = '宋国毛五贴五毛党国家是谁啊';

		let r = ac.search(content, function (found_word, data, offset)
		{
			expect(content.substr(offset, found_word.length))
				.to.deep.equal(found_word)
		});

		expect(find_list.length).to.ok;
		expect(Object.keys(r.count)).to.deep.equal(find_list)

		console.log(r);
	});
});
