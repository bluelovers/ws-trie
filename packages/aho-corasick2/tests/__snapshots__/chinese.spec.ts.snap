// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picks out chinese words 1`] = `
AhoCorasick {
  "trie": Trie {
    "data": [],
    "is_word": null,
    "next": {
      "五": Trie {
        "data": [],
        "is_word": null,
        "next": {
          "毛": Trie {
            "data": [],
            "is_word": null,
            "next": {
              "党": Trie {
                "data": [
                  undefined,
                ],
                "is_word": true,
                "next": {},
                "value": "五毛党",
              },
            },
            "value": "五毛",
          },
        },
        "value": "五",
      },
      "党": Trie {
        "data": [],
        "is_word": null,
        "next": {
          "国": Trie {
            "data": [
              undefined,
            ],
            "is_word": true,
            "next": {},
            "value": "党国",
          },
        },
        "value": "党",
      },
      "国": Trie {
        "data": [],
        "is_word": null,
        "next": {
          "家": Trie {
            "data": [
              undefined,
            ],
            "is_word": true,
            "next": {},
            "value": "国家",
          },
        },
        "value": "国",
      },
      "毛": Trie {
        "data": [],
        "is_word": null,
        "next": {
          "五": Trie {
            "data": [],
            "is_word": null,
            "next": {
              "贴": Trie {
                "data": [
                  undefined,
                ],
                "is_word": true,
                "next": {},
                "value": "毛五贴",
              },
            },
            "value": "毛五",
          },
        },
        "value": "毛",
      },
      "黑": Trie {
        "data": [],
        "is_word": null,
        "next": {
          "手": Trie {
            "data": [],
            "is_word": null,
            "next": {
              "党": Trie {
                "data": [
                  undefined,
                ],
                "is_word": true,
                "next": {},
                "value": "黑手党",
              },
            },
            "value": "黑手",
          },
        },
        "value": "黑",
      },
    },
    "value": null,
  },
}
`;

exports[`Picks out chinese words 2`] = `
AhoCorasick {
  "trie": Trie {
    "data": [],
    "fail": null,
    "is_word": null,
    "next": {
      "五": Trie {
        "data": [],
        "fail": null,
        "is_word": null,
        "next": {
          "毛": Trie {
            "data": [],
            "fail": Trie {
              "data": [],
              "fail": null,
              "is_word": null,
              "next": {
                "五": Trie {
                  "data": [],
                  "fail": [Circular],
                  "is_word": null,
                  "next": {
                    "贴": Trie {
                      "data": [
                        undefined,
                      ],
                      "fail": null,
                      "is_word": true,
                      "next": {},
                      "value": "毛五贴",
                    },
                  },
                  "value": "毛五",
                },
              },
              "value": "毛",
            },
            "is_word": null,
            "next": {
              "党": Trie {
                "data": [
                  undefined,
                ],
                "fail": Trie {
                  "data": [],
                  "fail": null,
                  "is_word": null,
                  "next": {
                    "国": Trie {
                      "data": [
                        undefined,
                      ],
                      "fail": Trie {
                        "data": [],
                        "fail": null,
                        "is_word": null,
                        "next": {
                          "家": Trie {
                            "data": [
                              undefined,
                            ],
                            "fail": null,
                            "is_word": true,
                            "next": {},
                            "value": "国家",
                          },
                        },
                        "value": "国",
                      },
                      "is_word": true,
                      "next": {},
                      "value": "党国",
                    },
                  },
                  "value": "党",
                },
                "is_word": true,
                "next": {},
                "value": "五毛党",
              },
            },
            "value": "五毛",
          },
        },
        "value": "五",
      },
      "党": Trie {
        "data": [],
        "fail": null,
        "is_word": null,
        "next": {
          "国": Trie {
            "data": [
              undefined,
            ],
            "fail": Trie {
              "data": [],
              "fail": null,
              "is_word": null,
              "next": {
                "家": Trie {
                  "data": [
                    undefined,
                  ],
                  "fail": null,
                  "is_word": true,
                  "next": {},
                  "value": "国家",
                },
              },
              "value": "国",
            },
            "is_word": true,
            "next": {},
            "value": "党国",
          },
        },
        "value": "党",
      },
      "国": Trie {
        "data": [],
        "fail": null,
        "is_word": null,
        "next": {
          "家": Trie {
            "data": [
              undefined,
            ],
            "fail": null,
            "is_word": true,
            "next": {},
            "value": "国家",
          },
        },
        "value": "国",
      },
      "毛": Trie {
        "data": [],
        "fail": null,
        "is_word": null,
        "next": {
          "五": Trie {
            "data": [],
            "fail": Trie {
              "data": [],
              "fail": null,
              "is_word": null,
              "next": {
                "毛": Trie {
                  "data": [],
                  "fail": [Circular],
                  "is_word": null,
                  "next": {
                    "党": Trie {
                      "data": [
                        undefined,
                      ],
                      "fail": Trie {
                        "data": [],
                        "fail": null,
                        "is_word": null,
                        "next": {
                          "国": Trie {
                            "data": [
                              undefined,
                            ],
                            "fail": Trie {
                              "data": [],
                              "fail": null,
                              "is_word": null,
                              "next": {
                                "家": Trie {
                                  "data": [
                                    undefined,
                                  ],
                                  "fail": null,
                                  "is_word": true,
                                  "next": {},
                                  "value": "国家",
                                },
                              },
                              "value": "国",
                            },
                            "is_word": true,
                            "next": {},
                            "value": "党国",
                          },
                        },
                        "value": "党",
                      },
                      "is_word": true,
                      "next": {},
                      "value": "五毛党",
                    },
                  },
                  "value": "五毛",
                },
              },
              "value": "五",
            },
            "is_word": null,
            "next": {
              "贴": Trie {
                "data": [
                  undefined,
                ],
                "fail": null,
                "is_word": true,
                "next": {},
                "value": "毛五贴",
              },
            },
            "value": "毛五",
          },
        },
        "value": "毛",
      },
      "黑": Trie {
        "data": [],
        "fail": null,
        "is_word": null,
        "next": {
          "手": Trie {
            "data": [],
            "fail": null,
            "is_word": null,
            "next": {
              "党": Trie {
                "data": [
                  undefined,
                ],
                "fail": Trie {
                  "data": [],
                  "fail": null,
                  "is_word": null,
                  "next": {
                    "国": Trie {
                      "data": [
                        undefined,
                      ],
                      "fail": Trie {
                        "data": [],
                        "fail": null,
                        "is_word": null,
                        "next": {
                          "家": Trie {
                            "data": [
                              undefined,
                            ],
                            "fail": null,
                            "is_word": true,
                            "next": {},
                            "value": "国家",
                          },
                        },
                        "value": "国",
                      },
                      "is_word": true,
                      "next": {},
                      "value": "党国",
                    },
                  },
                  "value": "党",
                },
                "is_word": true,
                "next": {},
                "value": "黑手党",
              },
            },
            "value": "黑手",
          },
        },
        "value": "黑",
      },
    },
    "value": null,
  },
}
`;

exports[`Picks out chinese words 3`] = `
{
  "count": {
    "五毛党": 1,
    "党国": 1,
    "国家": 1,
    "毛五贴": 1,
  },
  "data": {
    "五毛党": [
      undefined,
    ],
    "党国": [
      undefined,
    ],
    "国家": [
      undefined,
    ],
    "毛五贴": [
      undefined,
    ],
  },
  "matches": {
    "五毛党": [
      5,
    ],
    "党国": [
      7,
    ],
    "国家": [
      8,
    ],
    "毛五贴": [
      2,
    ],
  },
  "positions": {
    "2": [
      "毛五贴",
    ],
    "5": [
      "五毛党",
    ],
    "7": [
      "党国",
    ],
    "8": [
      "国家",
    ],
  },
}
`;
